rest/model/templates/service_methods.gotmpl

```diff
@@ -1,11 +1,12 @@
 
-func (m *{{.RestType}}) BuildFromService(t {{shortenpackage .ModelType }}) error {
+func {{.RestType}}BuildFromService(t {{shortenpackage .ModelType }}) *{{.RestType}} {
```

Comment:
This seems related to your first bullet point about making these functions take one value and return one value, is that true? It seems odd to me to make this change, i.e. make BuildFromService/ToService _not_ be methods. Why was it necessary to make them regular functions?

Reply:
It was so that the same template can be used regardless of if the field is a scalar type or a complex type. If this is a struct method, then I'll need to check the type of the field when populating the template which is super messy

Reply:
Will this require us to update our existing BuildFromService/ToService functions?

Reply:
No, it's not an interface at all so it can have any signature. We could even rename it if it's not clear what "service" means for these functions (I definitely have no idea)